### DISCUSSION TITLE
Is it possible to use GLX/WGL without the rest of OpenGL headers?

### DISCUSSION THREAD

**Discussion by MarkLagodych** [Dav1dde/glad#398] [General]
I am working on a hobby window management library (like SDL but focused on GUI) and I wish to add functionality for creating OpenGL contexts, without actual OpenGL loading. The reason is that I want to use GLX/WGL/EGL for creating GL contexts and let the users use whatever OpenGL loader, version and extensions they want.
However, the generated `<glad/glx.h>` header includes `<glad/gl.h>` which does not seem to do much: it only provides one type `GLubyte` for a single function definition.

```c
// Generated with: http://glad.sh/#api=glx%3D1.4&extensions=&generator=c&options=
// Line 450 of src/glx.c
typedef __GLXextFuncPtr (GLAD_API_PTR *PFNGLXGETPROCADDRESSPROC)(const GLubyte * procName);
```

The generated `<glad/wgl.h>` also includes `<glad/gl.h>` but I did not find what definitions from there are used.

The question is: is it possible to use WGL/GLX without `<glad/gl.h>`?

**Comment by Dav1dde** [comment]
Yeah you can use it without but you'll have to replace the include (manually) with something that provides the symbols instead or define it yourself. `glad/gl.h` is only included for the required symbols.

**Reply by MarkLagodych** [reply]
Thank you!
